src/main/java/services/RestoreFactory.java

```diff
@@ -351,6 +353,14 @@ private String getSyncToken(String username) {
         return storage.getMetaDataFieldForRecord(users.firstElement(), User.META_SYNC_TOKEN);
     }
 
+    // Device ID for tracking usage in the same way Android uses IMEI
+    private String getSyncDeviceId() {
+        if (asUsername == null) {
+            return DEVICE_ID_SLUG;
+        }
+        return String.format("%s|%s|as|%s", DEVICE_ID_SLUG, username, asUsername);
```

Comment:
Would ti be possible to use the user_id instead of username? Just that we've had edge cases before where a user get's deleted and another one created with the same username.